backend/src/controllers/RoleController.ts

```diff
@@ -0,0 +1,84 @@
+import asyncHandler from 'express-async-handler';
+import { Request, Response } from 'express';
+import Role from '../models/Role';
+
+/**
+ * @desc 	Get all the roles
+ * @route 	GET /api/roles/
+ */
+const getAllRoles = asyncHandler(async (req: Request, res: Response) => {
+    const roles = await Role.find();
+
+    res.status(200).json({
+        status: 'success',
+        results: roles.length,
+        data: {
+            roles
+        }
+    });
+});
+
+/**
+ * @desc 	Get a specific role
+ * @route 	GET /api/roles/:
+ */
+const getRole = asyncHandler(async (req: Request, res: Response) => {
+    const role = await Role.findById(req.params.id);
+
+    res.status(200).json({
+        status: 'success',
+        data: {
+            role
+        }
+    });
+});
+
+/**
+ * @desc 	Add a new role
+ * @route 	POST /api/roles/
+ */
+const addRole = asyncHandler(async (req: Request, res: Response) => {
+
+    const newRole = await Role.create(req.body)    
+
+    newRole.save()
```

Comment:
I'm fairly certain `Role.create` automatically saves the new role.

From the [docs](https://mongoosejs.com/docs/api.html#model_Model.create):
> Shortcut for saving one or more documents to the database. MyModel.create(docs) does new MyModel(doc).save() for every doc in docs.